src/OAuth/ITokenClient.cs

```diff
@@ -16,5 +16,27 @@ public partial interface ITokenClient
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
         Task<GetAccessTokenResponse> GetAccessTokenFromServicePrincipalAsync(string servicePrincipalKey, AccessKey accessKey, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Gets an access token given a jwt code, client id, redirect URI, client secret and code verifier. These values can be exported from the Laserfiche Developer Console. This is the client credentials flow that applies to service applications.
+        /// </summary>
+        /// <param name="code">Authorization code</param>
+        /// <param name="redirectUri">Authorization endpoint redirect URI</param>
+        /// <param name="clientId">OAuth application client ID</param>
+        /// <param name="clientSecret">OPTIONAL OAuth application client secret. Required for web apps.</param>
+        /// <param name="codeVerifier">OPTIONAL PKCE code verifier. Required for SPA apps.</param>
+        /// <returns>A response that contains an access token</returns>
+        Task<GetAccessTokenResponse> GetAccessTokenFromCode(string code, string redirectUri, string clientId, string clientSecret, string codeVerifier);

```

Comment:
if client secret and code verifier are optional, they should have default values so that user doesn't need to pass it into the function

Reply:
ok done and was able to successfully manually test again

![image](https://user-images.githubusercontent.com/103658932/175400101-1037d1fb-f467-4f36-bbf0-c32bf2cbc9a9.png)